routes/index.js

```diff
@@ -42,11 +48,33 @@ router.post("/signup", async function (req, res, next) {
     return;
   }
 
-  const hashedPassword = bycrypt.hashSync(password, 10);
+  const hashedPassword = bcrypt.hashSync(password, 10);
 
   await db.insert({ email, password: hashedPassword, nickname });
 
   res.redirect("/login");
 });
 
+router.post("/login", async function (req, res) {
+  const { email, password } = req.body;
+  const foundUser = await db.find({ email });
+  if (foundUser.length && await bcrypt.compare(password, foundUser[0].password)) {
+    req.session.nickname = foundUser[0].nickname;
+    res.redirect('/');
+  } else {
+    res.render('login', { error: '존재하지 않는 유저정보입니다.' });
```

Comment:
`res.render` 를 통해 템플릿을 넣어주는 방식은 대부분의 상황에서 문제 없이 동작하나 다음과 같은 치명적인 단점이 있습니다.

로그인 에러 메시지가 발생하는 경우, 해당 요청은 `POST` 메서드에서 발생하고,

`POST` 메서드에서 렌더링 되었기 때문에 이후 새로고침을 하더라도 `GET`이 아닌 `POST` 요청으로 계속 `login` 문서를 서버에 요청하게 됩니다.

이 경우 로그인 에러메시지가 새로고침을 하더라도 사라지지 않는 치명적인 문제가 발생합니다.

이는 회원가입 기능을 구현할 때도 동일하게 발생했던 문제인데 세션이나 쿼리스트링을 이용해서 `res.redirect`를 해야할 것 같습니다. 만약 세션을 이용한다면 다음과 같이 수정할 수 있을 것 같습니다.

먼저 `router.post('/login, ...)` 라우터 내부를 아래와 같이 수정합니다.

```suggestion
    req.session.loginError = "존재하지 않는 유저정보입니다.";
    res.redirect("/login");
```
<br/>

그리고 `router.get('/login', ...`) 라우터로 이동하여 다음과 같이 수정합니다.

```js
router.get("/login", function (req, res, next) {
  const loginError = req.session?.loginError;

  if (loginError) {
    delete req.session.loginError;
    res.render("login", { error: loginError });
    return;
  }

  res.render("login");
});
```

Reply:
session으로 옮겨서 작업하도록 하겠습니다~